packages/amplify-migration-codegen-e2e/src/helpers.ts

```diff
@@ -3,21 +3,164 @@ import {
+    '.amplify',
+    'migration',
+    'amplify',
+    'backend',
+    'function',
+    gen1FunctionName.split('-')[0],
+    'src',
+    'index.js',
+  );
+  const destinationPath = path.join(projRoot, 'amplify', 'function', gen1FunctionName.split('-')[0], 'handler.ts');
+  const content = await fs.readFile(sourcePath, 'utf8');
+
+  // Replace the first occurrence of 'event' with 'event: any'
+  const modifiedContent = content.replace(/(exports\.handler\s*=\s*async\s*\(\s*)event(\s*\))/, '$1event: any$2');
+
+  await fs.writeFile(destinationPath, modifiedContent, 'utf8');
+}
+
+export async function copyGen1Schema(projRoot: string): Promise<void> {
+  const gen1SchemaPath = path.join(projRoot, '.amplify', 'migration', 'amplify', 'backend', 'api', 'codegentest', 'schema.graphql');
+  const gen1Schema = await fs.readFile(gen1SchemaPath, 'utf-8');
+  const dataResourcePath = path.join(projRoot, 'amplify', 'data', 'resource.ts');
+  const dataResourceContent = await fs.readFile(dataResourcePath, 'utf-8');
+  const backendPath = path.join(projRoot, 'amplify', 'backend.ts');
+  let backendContent = await fs.readFile(backendPath, 'utf-8');
+
+  const schemaRegex = /"TODO: Add your existing graphql schema here"/;
+  const updatedContent = dataResourceContent.replace(schemaRegex, `\`${gen1Schema.trim()}\``);
+
+  const errorRegex = /throw new Error\("TODO: Add Gen 1 GraphQL schema"\);?\s*/;
+  const finalContent = updatedContent.replace(errorRegex, '');
+
+  await fs.writeFile(dataResourcePath, finalContent, 'utf-8');
+
+  const linesToAdd = `
+  const todoTable = backend.data.resources.cfnResources.additionalCfnResources['Todo'];
+  todoTable.addOverride('Properties.sseSpecification', { sseEnabled: false });
+  `;
+
+  backendContent += linesToAdd;
+  await fs.writeFile(backendPath, backendContent, 'utf-8');
+}
+
+async function setEnableGen2MigrationFeatureFlag(projectRoot: string): Promise<void> {
+  const cliJsonPath = path.join(projectRoot, 'amplify', 'cli.json');
+  const cliJson = await fs.readJSON(cliJsonPath);
+  if (!cliJson.features) {
+    cliJson.features = {};
+  }
+  if (!cliJson.features.graphqltransformer) {
+    cliJson.features.graphqltransformer = {};
+  }
+  cliJson.features.graphqltransformer.enablegen2migration = true;
+  await fs.writeJSON(cliJsonPath, cliJson, { spaces: 2 });
+}
+
+async function updatePackageJsonDependency(cwd: string, dependencyName: string, version: string): Promise<void> {
+  const packageJsonPath = path.join(cwd, 'package.json');
+  const packageJsonContent = await fs.readFile(packageJsonPath, 'utf-8');
+  const packageJson = JSON.parse(packageJsonContent);
+
+  packageJson.devDependencies = packageJson.devDependencies || {};
+  packageJson.devDependencies[dependencyName] = version;
+
+  const updatedContent = JSON.stringify(packageJson, null, 2);
+  await fs.writeFile(packageJsonPath, updatedContent, 'utf-8');
+}
+
+async function getAppSyncDataSource(apiId: string, dataSourceName: string, region: string) {
```

Comment:
this helpers.ts file is growing in size. you may consider splitting it to auth_helpers, data_helpers...or something similar.

Reply:
Best if we don't have any `helpers` files.

https://www.google.com/search?q=helper+code+smell